### PR TITLE
Solar glossary

### PR DESCRIPTION
**Pull request by dshulyak**

### CODE REVIEW

File: docs/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      run: run.pp
+      update: run.pp
+
+Tags
+------
+Used to create arbitrary groups of resources, later this groups will be
+used for different user operations.
+
+Handlers
+=========
```
Comment (bogdando) [inline docs/glossary.rst]: one '=' less plz

File: docs/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+Layer that responsible for action execution and tracking result.
+Currently handler specified in resource meta.yaml and used for all resource
+actions ::
+
+  handler: puppet
```
Comment (bogdando) [inline docs/glossary.rst]: anything to be put about puppetv2?

Comment (dshulyak) [inline docs/glossary.rst]: maybe glossary shouldnt have a lot of details about each available handler?
i think right way would be to include link to handler speficic doc

File: docs/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+Transports
+-----------
+Used in handlers to communicate with managed by solar hosts. List of transports
+should be added to a node. Transports will be added to a resource by means
+of transports id.
+
+Two different types of transports are used: run and sync.
+Run transport - reponsible for running command on remote host.
+Sync transport - uploads required information.
```
Comment (bogdando) [inline docs/glossary.rst]: rsync?

File: docs/glossary.rst [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+location_id
+------------
+Used in transport layer to find ip address of a node.
```
Comment (bogdando) [inline docs/glossary.rst]: an example please?

File: docs/glossary.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+Events
+========
+Used in solar to describe all possible transitions between resources changes.
+Each event allows to specify two points of transitions, condition of this
+transition and type of event.
+
+Right now we are supporting 2 types of events.
+
+1. Dependency
+Inserts edge between 2 changes.
```
Comment (bogdando) [inline docs/glossary.rst]: into the deploy graph?

File: docs/glossary.rst [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+Inserts change specified in reaction and makes edge between parent and child.
+
+Example ::
+
+  type: depends_on
+  parent: nova-db
+  parent_action: run
+  child: nova-api
+  child_action: run
+  state: success // condition
```
Comment (bogdando) [inline docs/glossary.rst]: please some examples for state: failure and partial success by a fixed rate?

File: docs/glossary.rst [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+
+Virtual resources/templates
+===========================
+Composition layer that allows to allows to:
```
Comment (bogdando) [inline docs/glossary.rst]: allows

File: docs/glossary.rst [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+them.
+
+Staged log
+-----------
+Based on user changes - solar will create log of staged changes.
+This log will be used later to build deployment plan.
+
+History
+--------
+After action that is related to change wille be executed - it will be moved to history with same uuid.
```
Comment (bogdando) [inline docs/glossary.rst]: will be

File: docs/glossary.rst (hunk 2 above)
Comment (pigmej) [inline docs/glossary.rst]: +1 for dshulyak there. Each handler should have separate section in docs.

File: docs/glossary.rst (hunk 4 above)
Comment (pigmej) [inline docs/glossary.rst]: I don't think so it's needed there. `location_id` is just used to traverse and find correct `ip`.

Though we should probably change that a bit. To 
```
Solar uses it to find a physical location of given resource to collect for example valid ip address for transport.
```

File: docs/glossary.rst (hunk 3 above)
Comment (dshulyak) [inline docs/glossary.rst]: not always, the idea is to use any available sync transport, currently it is either scp or rsync

File: docs/glossary.rst (hunk 6 above)
Comment (dshulyak) [inline docs/glossary.rst]: there is not way to control it with events right now, i will update doc when it will be added

**Pull request merged**